---
title: "Estimating Ethnicity in the iChip 1-6 Dataset"
author: "Translational Genomics Group"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: github_document
---

```{r setup, include=FALSE, message = FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(readr)
library(dplyr)
library(tidyr)
library(knitr)
library(ggplot2)
library(cluster)
library(tibble)
```

## Goals

The principle goal of this notebook is to provide an ancestry estimate for each subject in the iChip1t6 dataset dataset that can later be used for sub-setting to more homogenous populations. 

Ancestry estimation will be performed using the ADMIXTURE tool. Allele frequencies will first be learned in a supervised manner from labeled data provided by 1000 Genomes samples belonging to the EUR, AFR and EAS populations. Then ancestry proportions will then be estimated for the iChip 1 -6 samples. 

## 1. Preapre iChip 1 to 6 Dataset

### A. Download Immuno_BeadChip_11419691_B Strand Files for iChip 1 to 6 and Ref/Alt File
```{bash eval = FALSE, message = FALSE}
wget http://www.well.ox.ac.uk/~wrayner/strand/Immuno_BeadChip_11419691_B-b37-strand.zip -O temp.zip; unzip temp.zip; rm temp.zip &>/dev/null

wget http://www.well.ox.ac.uk/~wrayner/strand/RefAlt/Immuno_BeadChip_11419691_B-b37.strand.RefAlt.zip -O temp.zip; unzip temp.zip; rm temp.zip &>/dev/null
```

### B. Create List of SNPs to Flip
```{r}
strand <- read_tsv("immuno_beadchip_11419691_b-b37.strand", 
                   col_names = c("SNP", "Chr", "BP", "Match", "Strand", "Allele"))

strand %>%
  filter(Strand == "-") %>%
  select(SNP) %>%
  write_tsv("temp_flip_ichip1t6.tsv", col_names = FALSE)
```

### C. Flip SNPs to Convert TOP to Forward/Ref in Plink
```{bash}
plink \
--bfile ../1.split_cohorts/cohort_split_ichip1t6 \
--flip temp_flip_ichip1t6.tsv \
--make-bed \
--out temp_ichip1t6
```

### D. Set Ref/Alt Allele 
```{bash}
plink \
--bfile temp_ichip1t6 \
--reference-allele Immuno_BeadChip_11419691_B-b37.strand.RefAlt 2 \
--make-bed \
--out temp2_ichip1t6
```

### D. Download 1000G Check Tool 
```{bash eval = FALSE}
wget http://www.well.ox.ac.uk/~wrayner/tools/HRC-1000G-check-bim-v4.2.9.zip -O temp.zip; unzip temp.zip; rm temp.zip
```

### E. Download 1000 Genome Reference
This is a large file ~10GB and needs to be unzipped. A local copy has already been downloaded to: 
/mnt/share6/SHARED_DATASETS/1000_genomes/reference_file/1000GP_Phase3_combined.legend

```{bash, eval = FALSE}
wget http://www.well.ox.ac.uk/~wrayner/tools/1000GP_Phase3_combined.legend.gz ; gunzip -k 1000GP_Phase3_combined.legend.gz; rm 1000GP_Phase3_combined.legend.gz
```

### F. Filter to high quality SNPs For Ancestry Estimation
```{bash}
plink \
--bfile temp2_ichip1t6 \
--geno 0.03 \
--maf 0.01 \
--hwe 0.000001 \
--chr 1-22 \
--keep-allele-order \
--make-bed \
--out temp3_ichip1t6
```

### Calcualte Allele Frequency
```{bash}
plink \
--bfile temp3_ichip1t6 \
--keep-allele-order \
--freq \
--out temp3_ichip1t6
```

### Run 1000G Checks
This takes ~30 minutes. 
```{bash eval = FALSE}
perl HRC-1000G-check-bim.pl \
-b temp3_ichip1t6.bim \
-f temp3_ichip1t6.frq \
-r /mnt/share6/SHARED_DATASETS/1000_genomes/reference_file/1000GP_Phase3_combined.legend \
-g
```

#### Exlude SNPs 
```{bash}
plink2 \
--bfile temp3_ichip1t6 \
--exclude Exclude-temp3_ichip1t6-1000G.txt \
--set-all-var-ids @:#[hg19] \
--keep-allele-order \
--make-bed \
--out temp4_ichip1t6
```

## Process 1000 Genomes Data 
### Download Raw Data for Chr 1 - 22
```{bash eval = FALSE}
for i in {1..22}; do
       wget ftp://ftp-trace.ncbi.nih.gov/1000genomes/ftp/release/20130502/ALL.chr${i}.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz
done
```

### Get list of CHR and Position from Marker names to be used to extract subset of 1000 Genome
```{r message=FALSE}
snplist <- read_table2("temp4_ichip1t6.bim", col_names = c("Chr", "SNP", "Pos", "BP", "A1", "A2"))

snplist %>%
  select(Chr, BP) %>%
  write_tsv("temp_site_list.txt", col_names = FALSE)
```

### Extract Markers from 1000Genome with Chrom and Position that Match Share iCHIP
```{bash eval = FALSE}
for i in {1..22}; do
      vcftools --gzvcf /mnt/share6/SHARED_DATASETS/1000_genomes/ALL.chr${i}.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz \
               --positions temp_site_list.txt \
               --recode \
               --recode-INFO-all \
               --out ichip1t6_1000G_markers/chr${i}.1000g.ichip_sites &
done
```

### Concatenate chromosome VCFs into one VCF
```{bash eval = FALSE}
bcftools concat -f ../../original_data/ordered_filenames.txt -o ichip1t6_1000G_markers/chrall_1000g_ichip_sites.vcf
```

### Get List of 1000 Genome Samples Filtered for EUR, AFR, EAS superpopulation samples 
```{r message=FALSE}
onek_samples <- read_tsv("../../original_data/igsr_samples.tsv")
colnames(onek_samples) <- make.names(colnames(onek_samples))

onek_samples %>%
  filter(Superpopulation.code %in% c("EUR", "AFR", "EAS")) %>%
  mutate(FID = 0) %>%
  select(FID, Sample.name) %>% 
  write_tsv("temp_onek_samples_3_pops.tsv", col_names = FALSE)
```

### Read Combined Chromosomes 1000 Genomes Samples with only Share Markers into PLINK and Set Names and Exclude not 3 Population Samples
```{bash}
plink2 \
--vcf ichip1t6_1000G_markers/chrall_1000g_ichip_sites.vcf \
--set-all-var-ids @:#[hg19] \
--new-id-max-allele-len 23 'truncate' \
--keep temp_onek_samples_3_pops.tsv \
--make-bed \
--out temp_chrall_1000g_ichip_sites > /dev/null
```

## Merge iCHIP and 1000 Genomes
```{bash eval = FALSE}
plink \
--bfile temp4_ichip1t6 \
--bmerge temp_chrall_1000g_ichip_sites \
--merge-mode 6 \
--keep-allele-order \
--out ichip1t6_1000g_ichip_sites_merged
```

### Remove Non Bi-allelic SNPs iCHIP
```{bash}
plink \
--bfile temp4_ichip1t6 \
--exclude ichip1t6_1000g_ichip_sites_merged.missnp \
--make-bed \
--keep-allele-order \
--out temp5_ichip1t6
```

### Remove Non Bi-allelic SNPs 1000 Genome
```{bash}
plink \
--bfile temp_chrall_1000g_ichip_sites \
--exclude ichip1t6_1000g_ichip_sites_merged.missnp \
--make-bed \
--keep-allele-order \
--out temp1_chrall_1000g_ichip_sites
```

### Merge 
```{bash}
plink \
--bfile temp5_ichip1t6 \
--bmerge temp1_chrall_1000g_ichip_sites \
--keep-allele-order \
--out temp_ichip_1000g_sites_merged
```

## Prune Combined dataset
```{bash}
plink \
--bfile temp_ichip_1000g_sites_merged \
--exclude range ../../original_data/highLDregions.txt \
--indep 50 5 1.8 \
--geno .03 \
--maf .01 \
--keep-allele-order \
--out temp_ichip_1000g_sites_merged > /dev/null
```

#### Extarct Pruned SNP in Merged Dataset
```{bash}
plink \
--bfile temp_ichip_1000g_sites_merged \
--extract temp_ichip_1000g_sites_merged.prune.in \
--keep-allele-order \
--make-bed \
--out temp1_ichip_1000g_sites_merged
```

## Estimate Ancestry Using ADMIXTURE Supervised Learning with 4 1000 Genome Reference Populations
### Create .pop File for Identifying 1000 Genome Reference Samples
```{r warning=FALSE, message=FALSE}
merged_fam <- read_table2("temp1_ichip_1000g_sites_merged.fam", col_names = c("FAM", "IID", "ID_F", "ID_M", "SEX", "PHENOTYPE"), col_types = 
                            list(FAM = col_character(),
                                  IID = col_character(),
                                  ID_F = col_integer(),
                                  ID_M = col_integer(),
                                  SEX = col_integer(),
                                  PHENOTYPE = col_integer()))

merged_fam <- merged_fam %>%
  mutate(FAM = if_else(FAM == "0", IID, FAM))

merged_pop_file <- left_join(merged_fam, onek_samples, by = c("FAM" = "Sample.name")) 

merged_pop_file <- merged_pop_file %>%
  select(FAM, IID, Superpopulation.code) %>%
  mutate(Superpopulation.code = if_else(is.na(Superpopulation.code), "-", Superpopulation.code))

merged_pop_file %>%
  select(Superpopulation.code) %>%
  write_tsv("temp1_ichip_1000g_sites_merged.pop", col_names = FALSE)
```

### Run ADMIXTURE Supervised
This takes > 1 hour using 48 cores
```{bash eval = FALSE}
admixture temp1_ichip_1000g_sites_merged.bed 3 -j48 --supervised 
```

```{bash, eval = FALSE}
cp temp1_ichip_1000g_sites_merged.3.Q admix_ichip1t6_merged.3.Q
cp temp1_ichip_1000g_sites_merged.3.P admix_ichip1t6_merged.3.P
```

### Assign Clusters by KMeans, PAM, and >75%
```{r message=FALSE, warning=FALSE}
merged_admix <- read_table2("admix_ichip1t6_merged.3.Q", col_names = FALSE) 
set.seed(1234)
kmeans_out_merged <- kmeans(merged_admix, centers = 4, nstart = 100)

merged_admix_pop <- bind_cols(merged_admix, merged_pop_file)
merged_admix_pop <- merged_admix_pop %>%
  rename(EUR = X1, EAS = X2, AFR = X3) %>%
  mutate(kmeans_cluster = kmeans_out_merged$cluster) %>%
  mutate(cedars_75_class = if_else(EUR >= 0.75, "EUR",
                                    if_else(EAS >= 0.75, "EAS",
                                                    if_else(AFR >= 0.75, "AFR", "Mixed")))) %>%
  mutate(kmeans_class = if_else(kmeans_cluster == 1, "EUR",
                                if_else(kmeans_cluster == 2, "Mixed",
                                        if_else(kmeans_cluster == 3, "EAS",
                                                if_else(kmeans_cluster == 4, "AFR", "wilson")))))

merged_admix_pop %>%
  filter(Superpopulation.code == "-") %>%
  group_by(cedars_75_class) %>%
  summarise(n = n()) %>%
  kable()
  
merged_admix_pop %>%
  filter(Superpopulation.code == "-") %>%
  group_by(kmeans_class) %>%
  summarise(n = n()) %>%
  kable()


merged_admix_pop %>%
  filter(Superpopulation.code == "-") %>%
  arrange(desc(EUR), desc(AFR), desc(EAS)) %>%
  rownames_to_column() %>%
  mutate(rowname = as.integer(rowname)) %>%
  select(FAM, EUR, AFR, EAS, rowname) %>%
  gather(key = "ancestry", value = "proportion", -FAM, -rowname) %>%
  ggplot(aes(x = reorder(FAM, rowname), y = proportion, fill = ancestry)) + 
  geom_bar(stat = "identity") +
  scale_y_continuous(limits = c(NA, 1.05)) +
  theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank()) +
  labs(y = "Estimated Proportion", x = "Sample", title = "Estimated Ancestry Proportion of iChip1-6 Samples by Supervised Admixture") + scale_fill_discrete(name="Population")
  
  
  
```

### Perform PCA
```{bash}
plink \
--bfile temp1_ichip_1000g_sites_merged \
--pca header tabs \
--out temp1_ichip_1000g_sites_merged > /dev/null
```

### Plot PCAs
```{r}
merged_pca <- read_table2("temp1_ichip_1000g_sites_merged.eigenvec", col_types = cols(.default = "c")) 

merged_pca <- merged_pca %>%
  mutate(FID = ifelse(FID == "0", IID, FID)) %>%
  mutate(PC1 = as.double(PC1)) %>%
  mutate(PC2 = as.double(PC2))

merged_admix_pop_pca <- left_join(merged_admix_pop, merged_pca, by = c("FAM" = "FID", "IID" = "IID"))

merged_admix_pop_pca %>%
  filter(Superpopulation.code != "-") %>%
  ggplot(aes(x = PC1, y = PC2, color = as.factor(Superpopulation.code))) +
  geom_point() +
  labs(title = "PCA of 1000 Genome Samples (merged with iChip1-6) Colored by \n reported Super-Population") +
  theme(legend.title=element_blank())

merged_admix_pop_pca %>%
  ggplot(aes(x = PC1, y = PC2, color = as.factor(Superpopulation.code))) +
  geom_point() +
  labs(title = "PCA of 1000 Genome and iChip1-6 Samples Colored by \n Reported Super-Population") +
  theme(legend.title=element_blank())

merged_admix_pop_pca %>%
  filter(Superpopulation.code == "-") %>%
  ggplot(aes(x = PC1, y = PC2, color = as.factor(kmeans_class))) +
  geom_point() +
  labs(title = "PCA of iChip1-6 Samples Colored by KMeans") +
  theme(legend.title=element_blank())

merged_admix_pop_pca %>%
  filter(Superpopulation.code == "-") %>%
  ggplot(aes(x = PC1, y = PC2, color = as.factor(cedars_75_class))) +
  geom_point() +
  labs(title = "PCA of iChip1-6 Samples Colored by 75%") +
  theme(legend.title=element_blank())

```

### Write Assigned 
```{r}
merged_admix_pop_pca %>% 
  filter(Superpopulation.code == "-") %>%
  filter(cedars_75_class == "EUR") %>%
  select(FAM, IID) %>%
  write_tsv(path = "ichip1t6_eur_samples.tsv", col_names = FALSE)
  
```

### Split Cohort by Ethnicity
```{bash}
plink \
--bfile ../1.split_cohorts/cohort_split_ichip1t6 \
--keep ichip1t6_eur_samples.tsv \
--make-bed \
--out eur_cohort_split_ichip1t6
```

## Clean Up
```{r}
file.remove(list.files(pattern = "^temp", full.names = TRUE))
```
